--- im-chooser-1.6.4/libimchooseui/eggsmclient-dbus-gnome.c.orig	2012-11-23 07:17:43.000000000 +0100
+++ im-chooser-1.6.4/libimchooseui/eggsmclient-dbus-gnome.c	2013-07-26 17:24:57.550140470 +0200
@@ -77,7 +77,7 @@ sm_client_gdbus_get_interface_info(void)
 										 &err);
 
 		if (err) {
-			g_warning(err->message);
+			g_warning("%s", err->message);
 			return NULL;
 		}
 		info = g_dbus_interface_info_ref(introspection_data->interfaces[0]);
@@ -120,7 +120,7 @@ sm_client_dbus_gnome_end_session(EggSMCl
 	if (value)
 		g_variant_unref(value);
 	if (err) {
-		g_warning(err->message);
+		g_warning("%s", err->message);
 		g_error_free(err);
 
 		return FALSE;
@@ -160,7 +160,7 @@ egg_sm_client_dbus_gnome_new(void)
 					      NULL,
 					      &err);
 	if (err) {
-		g_warning(err->message);
+		g_warning("%s", err->message);
 		g_error_free(err);
 		return NULL;
 	}
--- im-chooser-1.6.4/libimchooseui/eggsmclient-dbus-kde.c.orig	2012-11-23 07:17:52.000000000 +0100
+++ im-chooser-1.6.4/libimchooseui/eggsmclient-dbus-kde.c	2013-07-26 17:25:23.571293078 +0200
@@ -77,7 +77,7 @@ sm_client_kdbus_get_interface_info(void)
 										 &err);
 
 		if (err) {
-			g_warning(err->message);
+			g_warning("%s", err->message);
 			return NULL;
 		}
 		info = g_dbus_interface_info_ref(introspection_data->interfaces[0]);
@@ -112,7 +112,7 @@ sm_client_dbus_kde_end_session(EggSMClie
 	if (value)
 		g_variant_unref(value);
 	if (err) {
-		g_warning(err->message);
+		g_warning("%s", err->message);
 		g_error_free(err);
 
 		return FALSE;
@@ -152,7 +152,7 @@ egg_sm_client_dbus_kde_new(void)
 					      NULL,
 					      &err);
 	if (err) {
-		g_warning(err->message);
+		g_warning("%s", err->message);
 		g_error_free(err);
 		return NULL;
 	}
--- im-chooser-1.6.4/src/app/main.c.orig	2013-04-12 09:55:59.000000000 +0200
+++ im-chooser-1.6.4/src/app/main.c	2013-07-26 17:30:37.898551681 +0200
@@ -92,7 +92,7 @@ _show_error_cb(IMChooseUI  *ui,
 						 GTK_DIALOG_MODAL,
 						 GTK_MESSAGE_ERROR,
 						 GTK_BUTTONS_OK,
-						 s);
+						 "%s", s);
 	logfile = g_build_filename(g_get_user_cache_dir(), "imsettings", "log", NULL);
 	m = g_strdup_printf(_("Please check <a href=\"file://%s\">%s</a> for more details"),
 			    logfile, logfile);
--- im-chooser-1.6.4/src/xfce/main.c.orig	2012-11-23 07:23:19.000000000 +0100
+++ im-chooser-1.6.4/src/xfce/main.c	2013-07-26 17:32:39.190592588 +0200
@@ -94,8 +94,8 @@ _show_error_cb(IMChooseUI  *ui,
 						 GTK_DIALOG_MODAL,
 						 GTK_MESSAGE_ERROR,
 						 GTK_BUTTONS_OK,
-						 s);
-	gtk_message_dialog_format_secondary_text(GTK_MESSAGE_DIALOG (dlg), details);
+						 "%s", s);
+	gtk_message_dialog_format_secondary_text(GTK_MESSAGE_DIALOG (dlg), "%s", details);
 
 	/* for GNOME HIG compliance */
 	gtk_rc_parse_string("style \"imchoose-ui-message-dialog\" {\n"	\
